yt/visualization/accumulators.py

```diff
@@ -0,0 +1,348 @@
+    these, the user can compute the integral of any field in the dataset
+    along the given paths. The results of these path integrals are
+    stored in a cumulative fashion, which means that, once computed, the
+    accumulated value of the field is stored at each point along the
+    path, allowing the user to query this information.
+
+    Attributes
+    ----------
+    pass
+
+    Methods
+    -------
+    pass
+
+    Examples
+    --------
+    pass
+    """
+    def __init__(self, paths, ds):
+        self.paths      = []
+        self.ds         = ds
+        self.ad         = ds.all_data()
+        self.accum      = []
+        self.left_edge  = self.ds.domain_left_edge
+        self.right_edge = self.ds.domain_right_edge
+        # Make sure that the path has proper units. If no units are
+        # specified, assume code units
+        for p in paths:
+            if not isinstance(p, YTArray) or p.units == unyt.dimensionless:
+                self.paths.append(self.ds.arr(p, 'code_length'))
+            else:
+                self.paths.append(p.to(self.ds.length_unit))
+                    
+                
+                
+
+    def _get_tree(self, field, is_vector):
+        r"""
+        Creates an AMRKDTree and then adds the field to it.
+
+        Parameters
+        ----------
+        field : list 
+            Either the scalar field to add to the tree or an iterable
+            containing the components of the vector field to add.
+
+        is_vector : bool
+            If True, then `field` is a vector and it is assumed that it's
+            an iterable containing the x, y, and z components of the field.
+            Otherwise, the field is assumed to be a scalar field.
+
+        Returns
+        -------
+        tree : AMRKDTree
+            An AMRKDTree for the dataset containing the given field values.
+        """
+        tree = AMRKDTree(self.ds)
+        use_log = [False for i in range(len(field))]
+        field = [self.ad._determine_fields(f)[0] for f in field]
+        tree.set_fields(field, use_log, False)
+        return tree
+
+    def _get_path_field_values(self, tree, path, idx, vals, npts):
+        r"""
+        Determines the value of the field at each point along the path.
+
+        Parameters
+        ----------
+        tree : AMRKDTree
+            An AMRKDTree for the dataset containing the given field values.
+
+        path : np.ndarray
+            The N x 3 array containing the x, y, and z coordinates of the
+            N points used to define the path.
+
+        Returns
+        -------
+        path_field_values : np.ndarray
+            The array containing the field values at each point along the
+            path.
+        """
+        node = tree.locate_node(path[idx])
+        # Data is a list, with one element for each field component
+        data = node.data.my_data
+        # Number of cells in the node
+        ncells = data[0].shape
```

Comment:
Maybe change the name of `ncells` to something like `dims` to indicate that it's not just number of cells, but cells-by-dimension.

Reply:
Done.